script/update-clang.py

```diff
@@ -0,0 +1,240 @@
+LLVM_BUILD_DIR = os.path.realpath(LLVM_DIR)
+STAMP_FILE = os.path.join(LLVM_BUILD_DIR, 'cr_build_revision')
+
+# URL for pre-built binaries.
+CDS_URL = os.environ.get('CDS_CLANG_BUCKET_OVERRIDE',
+    'https://commondatastorage.googleapis.com/chromium-browser-clang')
+
+# Bump after VC updates.
+DIA_DLL = {
+  '2013': 'msdia120.dll',
+  '2015': 'msdia140.dll',
+  '2017': 'msdia140.dll',
+}
+
+
+def DownloadUrl(url, output_file):
+  """Download url into output_file."""
+  CHUNK_SIZE = 4096
+  TOTAL_DOTS = 10
+  num_retries = 3
+  retry_wait_s = 5  # Doubled at each retry.
+
+  while True:
+    try:
+      sys.stdout.write('Downloading %s ' % url)
+      sys.stdout.flush()
+      response = urllib2.urlopen(url)
+      total_size = int(response.info().getheader('Content-Length').strip())
+      bytes_done = 0
+      dots_printed = 0
+      while True:
+        chunk = response.read(CHUNK_SIZE)
+        if not chunk:
+          break
+        output_file.write(chunk)
+        bytes_done += len(chunk)
+        num_dots = TOTAL_DOTS * bytes_done / total_size
+        sys.stdout.write('.' * (num_dots - dots_printed))
+        sys.stdout.flush()
+        dots_printed = num_dots
+      if bytes_done != total_size:
+        raise urllib2.URLError("only got %d of %d bytes" %
+                               (bytes_done, total_size))
+      print ' Done.'
+      return
+    except urllib2.URLError as e:
+      sys.stdout.write('\n')
+      print e
+      if num_retries == 0 or isinstance(e, urllib2.HTTPError) and e.code == 404:
+        raise e
+      num_retries -= 1
+      print 'Retrying in %d s ...' % retry_wait_s
+      time.sleep(retry_wait_s)
+      retry_wait_s *= 2
+
+
+def EnsureDirExists(path):
+  if not os.path.exists(path):
+    print "Creating directory %s" % path
+    os.makedirs(path)
+
+
+def DownloadAndUnpack(url, output_dir):
+  with tempfile.TemporaryFile() as f:
+    DownloadUrl(url, f)
+    f.seek(0)
+    EnsureDirExists(output_dir)
+    tarfile.open(mode='r:gz', fileobj=f).extractall(path=output_dir)
+
+
+def ReadStampFile(path=STAMP_FILE):
+  """Return the contents of the stamp file, or '' if it doesn't exist."""
+  try:
+    with open(path, 'r') as f:
+      return f.read().rstrip()
+  except IOError:
+    return ''
+
+
+def WriteStampFile(s, path=STAMP_FILE):
+  """Write s to the stamp file."""
+  EnsureDirExists(os.path.dirname(path))
+  with open(path, 'w') as f:
+    f.write(s)
+    f.write('\n')
+
+
+def RmTree(directoryPath):
+  """Delete dir."""
+  def ChmodAndRetry(func, path, _):
+    # Subversion can leave read-only files around.
+    if not os.access(path, os.W_OK):
+      os.chmod(path, stat.S_IWUSR)
+      return func(path)
+    raise
+
+  shutil.rmtree(directoryPath, onerror=ChmodAndRetry)
+
+
+def CopyFile(src, dst):
+  """Copy a file from src to dst."""
+  print "Copying %s to %s" % (src, dst)
+  shutil.copy(src, dst)
+
+
+vs_version = None
+def GetVSVersion():
```

Comment:
Is this needed? I can't see this referenced anywhere else in the code

Reply:
We can  remove it. It will not break anything.

Whole update_clang.py has been taken from Chromium. Originally I did not wanted to modify it but I had to change the code which detects which version of Visual Studio is used.
I still hope, in the future, we will be able to use more advanced code from Chromium to do this. So I have left function GetVSVersion() intact to remind me, which function should be used for real VS detection.

Reply:
I hope that this issue will go away once we "merge" libcc and Electron repos.

Reply:
I'd rather have an issue as a reminder instead of essentially dead code that's never called TBH, that's a bit less confusing.

Reply:
But if this is critical for you guys to merge, I'm happy to give it a green light with this left in too.

Reply:
@torycl Let's remove the dead code and create an issue instead in the `e/e` issue tracker.